doc/source/io.rst

```diff
@@ -4419,44 +4419,6 @@ Now you can import the ``DataFrame`` into R:
    starting point if you have stored multiple ``DataFrame`` objects to a
    single HDF5 file.
 
-Backwards Compatibility
-'''''''''''''''''''''''
-
-0.10.1 of ``HDFStore`` can read tables created in a prior version of pandas,
-however query terms using the
-prior (undocumented) methodology are unsupported. ``HDFStore`` will
-issue a warning if you try to use a legacy-format file. You must
-read in the entire file and write it out using the new format, using the
-method ``copy`` to take advantage of the updates. The group attribute
-``pandas_version`` contains the version information. ``copy`` takes a
-number of options, please see the docstring.
-
-
-.. ipython:: python
-   :suppress:
-
-   import os
-   legacy_file_path = os.path.abspath('source/_static/legacy_0.10.h5')
-
```

Comment:
actually you can also remove the legacy stuff from the tests as well (and add a small note in the whatsnew)

Reply:
Ok.

Is it correct we're talking about the ``tests/io/test_pytables.py::testHDFStore.test_legacy*`` functions? (4 tests)

Reply:
Can we leave test changes for a separate PR please?

Reply:
I've made a start to it (#17398), but I have some issuse.

One issue is that this doc fragment uses a binary hdf file that needs to be deleted. Can this request be accepted now, so the other will pass? I will do it today and/or tomorrow.

Reply:
you can remove this part of the doc in the other PR